### ISSUE TITLE
SIGSEGV crash on new support for touch screens

### ISSUE THREAD

**Issue by seleuco** [mamedev/mame#12429]
### MAME version

0.266 (also on 0.265)

### System information

it is not relevant (cross platform issue)

### INI configuration details

```INI
UI.ini should have:

last_used_filter          Available
```

### Emulated system/software

MAME frontend

### Incorrect behaviour

SIGSEGV crash when you start MAME without any ROMs files and the available filter is default selected and you move your finger in the roms list where there is not any entry.

### Expected behaviour

Don't crash...

### Steps to reproduce

1. Start MAME without any ROMS files and the available filter selected previusly
2. Move your finger on ROM's panel

### Additional details

Hi DEVs,  I'm  the developer who maintains the MAME port for android.

Since I added the new touchscreen support added in the mame version 0.265, I have observed in the developer console how the telemetry is indicating an increase in errors in users exceeding the desirable threshold.

It is incredible how few errors I have seen untile now in the telemetry in software as large and complex as MAME. Most of the errors are from Android itself and its execution environment :)

Investigating this error I have observed that it has occurred since version 0.265 and when I added the new touch support for the frontend on Android. 

I have debugged the error a bit and have seen that the error may occur on menu_select_game::get_selection used in menu_select_launch::arts_render. It's seems its returning an ui_system_info which have a bad driver pointer becouse m_prev_selected it's not setted correctly.

It is important to keep in mind that in a new installation on Android, the availability filter is selected automatically by default so that the frontend to be less intimidating to new users and since the user does not have roms and tend to swipe the finger on the empty list, the error is occurring a lot.

I preferred to open this issue to see if you can take a look at it so I don't make an ugly hack...

This is the stack:

```
pid: 0, tid: 18149 >>> com.seleuco.mame4d2024 <<<

backtrace:
  #00  pc 0x000000000ee19ea8  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (ui::menu_select_launch::arts_render(unsigned int)+368) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #01  pc 0x000000000ee195b4  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (ui::menu_select_launch::draw_right_panel(unsigned int)+276) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #02  pc 0x000000000ee18b74  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (ui::menu_select_launch::draw(unsigned int)+492) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #03  pc 0x000000000edbf7e0  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (ui::menu::global_state::ui_handler(render_container&)+928) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #04  pc 0x000000000ee4cc2c  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (mame_ui_manager::update_and_render(render_container&)+676) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #05  pc 0x0000000013b8bbf4  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (video_manager::frame_update(bool)+136) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #06  pc 0x0000000013af2ad4  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (running_machine::run(bool)+796) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #07  pc 0x000000000e68d510  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (mame_machine_manager::execute()+468) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #08  pc 0x000000000ed8bf88  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (cli_frontend::start_execution(mame_machine_manager*, std::__ndk1::vector<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>, std::__ndk1::allocator<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>>> const&)+504) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #09  pc 0x000000000ed8d490  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (cli_frontend::execute(std::__ndk1::vector<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>, std::__ndk1::allocator<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>>>&)+60) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #10  pc 0x000000000e68e710  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (emulator_info::start_frontend(emu_options&, osd_interface&, std::__ndk1::vector<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>, std::__ndk1::allocator<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>>>&)+44) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #11  pc 0x000000000a78fab0  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (myosd_main+276) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #12  pc 0x000000000a78ebf8  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/lib/arm64/libMAME4droid.so (myosd_droid_main+3868) (BuildId: 7e21159c0d6c243f4dce07537d731bd673d06f10)
  #13  pc 0x0000000000351e30  /apex/com.android.art/lib64/libart.so (art_quick_generic_jni_trampoline+144)
  #14  pc 0x00000000005b8b18  /apex/com.android.art/lib64/libart.so (nterp_helper+152)
  #15  pc 0x000000000000af76  /data/app/~~Ik2MtufunI7u6_42ai7fSw==/com.seleuco.mame4d2024-S_2Ym2v_Z80DBFfltGUPMw==/base.apk (com.seleuco.mame4droid.Emulator$4.run+1290)
  #16  pc 0x000000000041c820  /data/misc/apexdata/com.android.art/dalvik-cache/arm64/boot.oat (java.lang.Thread.run+64)
  #17  pc 0x000000000033b3a4  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+612)
  #18  pc 0x000000000023a5d0  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+144)
  #19  pc 0x0000000000539bf4  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1600)
  #20  pc 0x00000000005395a4  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallbackWithUffdGc(void*)+8)
  #21  pc 0x00000000000fd0f4  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+208)
  #22  pc 0x0000000000096a04  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+68)
```

**Comment by cuavas** [comment]
Fixed by 9af617c8eeea2e0e2fcc28b526d21251b746e522.

The part that actually fixes the crash is this line:
```diff
--- a/src/frontend/mame/ui/selmenu.cpp
+++ b/src/frontend/mame/ui/selmenu.cpp
@@ -3157,7 +3157,7 @@ bool menu_select_launch::main_force_visible_selection()
 			return true;
 		}
 	}
-	else
+	else if (m_prev_selected)
 	{
 		int selection(0);
 		while ((m_available_items > selection) && (item(selection).ref() != m_prev_selected))
```

It shouldn’t be trying to ensure a visible primary item is selected if there are no primary visible items.

The changes to `menu_select_launch::draw` fix a cosmetic issue, and the rest of it is minor stuff that doesn’t cause functional changes.

**Comment by seleuco** [comment]
Thanks. Tested and works great :)
